### PR TITLE
Style consistency pass on Spatial, Fog, CanvasItem, Sky, Particle shader pages

### PR DESCRIPTION
**Pull request by tetrapod00**
Resolves https://github.com/godotengine/godot-docs/issues/9745
Also looking at the text in https://github.com/godotengine/godot-docs/pull/9691 as a guide for some of these changes.

Formalized style guidelines:
- Use sentence case for built-in descriptions. Avoid random capitalizations and title case.
- Shader built-ins should always be styled as code: `VERTEX`.
- Built-in bool descriptions should start with "`true` if...", "`true` when...", etc.
- Literal values should almost always be styled as code: `1.0`, `false`, `ENUM_VALUE`. Ints are `1`. Floats are styled as `1.0`.
- Processor functions should usually be styled as "the `light()` function" in the body and in built-in tables. Use of "the light function" or "a light function" are also okay, if it improves readability. Avoid "the light() function" (without code style).
- Always capitalize "UV".
- When referring to classes or properties, link to the class reference if possible.
- Use "model space" instead of "local space" or "local coordinates". Exception: refer to "model/local space" at least a couple times to establish the connection. Exception: CanvasItem page consistently uses "local space" and I left it alone, since I'm not deeply familiar with 2D spaces.
- https://github.com/godotengine/godot-docs/pull/9757 may let us standardize completely on "model space" except for in the definition table for coordinate spaces.
- In built-in descriptions, try to describe the space as "in world space". For example: NODE_POSITION_WORLD is "Node position, in world space", not "Node world space position".
- Render modes should be stylized as "using the `unshaded` render mode". Avoid "render_mode" in the body of the page.

Other, opinionated, changes made by this PR:
- In description of `VERTEX`, call it "vertex position" where possible. ctrl+f "position" should surface VERTEX.
- Similarly, get the word "color" into `ALBEDO`'s description somehow.
- Link to [Standard Material 3D](https://docs.godotengine.org/en/stable/tutorials/3d/standard_material_3d.html) from Spatial shaders. It's the best documentation for alpha modes among other things.
- A few small style, clarity, or grammar changes

### CODE REVIEW

**Comment by clayjohn** [comment]
Looks great so far! I agree with your style guideline

**Comment by tetrapod00** [comment]
Does Godot have a standard term for local/model space? Within the [Spatial Vertex Built-ins table](https://docs.godotengine.org/en/latest/tutorials/shaders/shader_reference/spatial_shader.html#vertex-built-ins), I see both "local coordinates" and "model space". 

I suppose if both are commonly used, it's better to keep references to both in the docs?

**Comment by tetrapod00** [comment]
"Local space" or "local coordinates" does match the naming convention for local coordinates in CPU-side transforms. If we standardize on "model space", there should still be a few mentions of "model / local space" to make the connection.

At some point all the different spaces should be documented in one place, so that single explanation can be linked to.

Another complication: the CanvasItem page uses "local space" as the main description. Should a different standard be applied to CanvasItem shaders since they are more of an on-ramp for new shader users?

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (tetrapod00) [inline tutorials/shaders/shader_reference/spatial_shader.rst:15]: Would like suggestions on the best wording for this. We do need to link to the Standard Material page, since it's the best/only documentation for many features of spatial shaders. I had it in the opening paragraph for a while. Putting the link here keeps it at the top of the page, while being directly relevant to the section.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:264]: All the references to "Light" I left alone. I am not familiar with 2D shaders - do these refer to the `LIGHT` built-in, or a 2D light node?

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:170]: Unsure about this change. The CanvasItem page seems more beginner friendly than the Spatial page. "Origin" is a term that may be unfamiliar.

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (tetrapod00) [inline tutorials/shaders/shader_reference/spatial_shader.rst:340]: Unconfirmed, based on the description of `skip_vertex_transform` and `world_vertex_coords`.

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (tetrapod00) [inline tutorials/shaders/shader_reference/spatial_shader.rst:342]: Unconfirmed, based on the description of `skip_vertex_transform` and `world_vertex_coords`.

**Comment by tetrapod00** [comment]
This one's ready for review. 

I can also clean up the guidelines themselves if we think we'll be reusing them. I think it's a fine outcome if these pages get cleaned up once and we don't formalize the style guidelines anywhere.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (huwpascoe) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:9]: "contain fewer"

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (huwpascoe) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:170]: position in window or in viewport?

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (huwpascoe) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:244]: again, is that window or viewport?

File: tutorials/shaders/shader_reference/particle_shader.rst
Comment (huwpascoe) [inline tutorials/shaders/shader_reference/particle_shader.rst:110]: Would "USERDATA#" be better?

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (huwpascoe) [reply tutorials/shaders/shader_reference/spatial_shader.rst:13]: Seems okay!

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (huwpascoe) [reply tutorials/shaders/shader_reference/spatial_shader.rst:340]: These would be in view space, yes.

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (huwpascoe) [reply tutorials/shaders/shader_reference/spatial_shader.rst:342]: 👍

File: tutorials/shaders/shader_reference/sky_shader.rst
Comment (huwpascoe) [inline tutorials/shaders/shader_reference/sky_shader.rst:175]: Maybe would be better calling these "LIGHT#"

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (huwpascoe) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst:265]: Refers to a 2D light node, (technically a "light instance" I think)

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst:170]: I'm not that familiar with 2d coordinate spaces and shaders, I'll investigate this and clarify

File: tutorials/shaders/shader_reference/sky_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/sky_shader.rst:175]: It's a little more readable in the docs, sure.

A side effect is that if you paste "LIGHT#_ENABLED" directly, you get an `Invalid symbols placed before directive` error. If you paste "LIGHTX_ENABLED" directly, you get an `Unknown identifier in expression: 'LIGHTX_COLOR'` error. So the error is more readable if we leave this as is, and will catch some nonzero amount of copy paste errors.

File: tutorials/shaders/shader_reference/particle_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/particle_shader.rst:110]: See review for LIGHTX

File: tutorials/shaders/shader_reference/sky_shader.rst
Comment (clayjohn) [reply tutorials/shaders/shader_reference/sky_shader.rst:175]: I don't have any preference here. So I will leave it up to tetrapod00

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (clayjohn) [reply tutorials/shaders/shader_reference/spatial_shader.rst:13]: Yep, seems fine

**Review by clayjohn — APPROVED** [review]
Seems great to me. There are a few outstanding comments from [huwpascoe](https://github.com/huwpascoe)'s review. Should be good to merge once those are resolved.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst:170]: It seems to be based on the viewport. Here's a quick fragcoord visualization shader, applied on both a full rect control and a subviewport. The subviewport's FRAGCOORD does not change as its position in the window changes.
Godot_v4 3-stable_win64_Ae2JA0ufjW

https://github.com/user-attachments/assets/b3d93a24-b182-4ec6-8e6a-712ecbeef857

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst:244]: The origin for `FRAGCOORD` in `light()` might be the upper left, too? This shader which outputs the `light()` `FRAGCOORD` seems to increase down and to the left, just like the `FRAGCOORD` in `fragment()`.

Godot_v4 3-stable_win64_UpaFokh9d3
Godot_v4 3-stable_win64_btnYCzoCrS

The setup for testing a subviewport is more complicated for Node2D, still working on that.

File: tutorials/shaders/shader_reference/sky_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/sky_shader.rst:175]: Leaving these as-is, erring on the side of fewer changes, and because of the the minor improvement in error clarity.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst:244]: Conclusion: the `light()` `FRAGCOORD` is also the coordinate in the viewport, not the window. And it's origin is the upper-left, like the `fragment()` `FRAGCOORD`. 
Godot_v4 3-stable_win64_kPGb6eQdcU

https://github.com/user-attachments/assets/8715d983-3174-4695-a1ad-6dda94ae0cd7

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst]: Changed these into links to Light2D and it's properties.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst]: Updated the description for both instances of FRAGCOORD to match observed behavior. This still might be technically a little incorrect, though?

**Comment by tetrapod00** [comment]
I made another round of changes for all of the outstanding comments. Would appreciate another correctness check on the changes!

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (huwpascoe) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst]: I figured the original incorrectly stated window instead of viewport.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (AThousandShips) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:265]:
```suggestion
|                                  | If the light is a :ref:`class_PointLight2D`, multiplied by the light's       |
```

File: tutorials/shaders/shader_reference/particle_shader.rst
Comment (AThousandShips) [inline tutorials/shaders/shader_reference/particle_shader.rst:161]:
```suggestion
| in vec3 **COLLISION_NORMAL**       | A normal of the last collision. If there is no collision detected it is equal to ``(0.0,0.0,0.0)``. |
```
```suggestion
| in vec3 **COLLISION_NORMAL**       | A normal of the last collision. If there is no collision detected it is equal to ``(0.0, 0.0, 0.0)``. |
```
And adjust the whole table

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (AThousandShips) [inline tutorials/shaders/shader_reference/spatial_shader.rst:396]:
```suggestion
| out vec4 **RADIANCE**                  | If written to, blends environment map radiance with ``RADIANCE.rgb`` based on ``RADIANCE.a``.    |
```

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (AThousandShips) [inline tutorials/shaders/shader_reference/spatial_shader.rst:470]:
```suggestion
|                                   | property for :ref:`class_OmniLight3D` and :ref:`class_SpotLight3D`.    |
```

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (AThousandShips) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:223]:
```suggestion
Use the ``unshaded`` render mode if you do not want the ``light()`` function to
```
Or keep the underscore IMO, this reads ungrammatical to me

File: tutorials/shaders/shader_reference/particle_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/particle_shader.rst:161]: I prefer `vec3(0.0, 0.0, 0.0)` for the literal, to make the type clear. I'll make sure to use the space between x, y, z, w for any vector literals in these pages though.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst:223]: I think either "the render mode ``unshaded``", or "the ``unshaded render mode" are good. But "render mode ``unshaded``" without a "the" is bad. I'd like to avoid "render_mode" without code styling in body text.

File: tutorials/shaders/shader_reference/particle_shader.rst
Comment (tetrapod00) [reply tutorials/shaders/shader_reference/particle_shader.rst:161]: Actually, looking at the pages again, you're right that just ``(0.0, 0.0, 0.0)`` is better. In other cases, writing the vector literals without the `vec3`, `vec2` constructor looks better, and in this case it could go either way. So I'll default to no constructor.

File: tutorials/shaders/shader_reference/canvas_item_shader.rst
Comment (AThousandShips) [reply tutorials/shaders/shader_reference/canvas_item_shader.rst:223]: "the render mode `unshaded`" is ungrammatical I'd say, so pick the swapped order

**Comment by tetrapod00** [comment]
Besides the resolved comments, I also changed how ranges were described in most cases in the spatial shader built-in tables.
Before, a range between 0 and 1 was described as "(0..1)". Now, it's "range of `[0.0, 1.0]`".
- Use code styling for the literals.
- Use square brackets to describe range, to avoid confusion with vector literals. 
- Explicitly mention "range of". It's wordy, but increases clarity.

**Comment by tetrapod00** [comment]
Oh, forgot to resolve a couple entries in the spatial shaders light() table.

File: tutorials/shaders/shader_reference/spatial_shader.rst
Comment (tetrapod00) [inline tutorials/shaders/shader_reference/spatial_shader.rst:465]: These two were awkward to write because you can't have links in a code block. I think I found a decent compromise.
firefox_fSIRLFZFLl

**Comment by skyace65** [comment]
Thanks!

**Pull request merged**
